kotlin/codegen/src/test/java/com/squareup/moshi/kotlin/codegen/JsonClassCodegenProcessorTest.kt

```diff
@@ -398,6 +398,37 @@ class JsonClassCodegenProcessorTest {
     )
   }
 
+  @Test
+  fun `Properties with @Json(isTransient = true) should be ignored`() {
```

Comment:
This should just be a functional test, that way we can make sure it actually works too.

Are there any other cases we should check with this?